sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneSharedAccessSignatureTokenProvider.cs

```diff
@@ -56,7 +56,11 @@ public TrackOneSharedAccessTokenProvider(SharedAccessSignature sharedAccessSigna
             Guard.ArgumentNotNullOrEmpty(nameof(resource), resource);
             Guard.ArgumentNotNegative(nameof(tokenValidityDuration), tokenValidityDuration);
 
-            if (!String.Equals(resource, SharedAccessSignature.Resource, StringComparison.InvariantCultureIgnoreCase))
+            // The requested resource may be a subpath of the current token resource, if a specific partition or other
+            // child object is referenced.  In that caes, the resource of the token should correspond to the starting
+            // segment of the requested resource.
+
+            if (resource.IndexOf(SharedAccessSignature.Resource, StringComparison.InvariantCultureIgnoreCase) != 0)
```

Comment:
I don't understand this check -- are we trying to enforce that the resource is the first entry in the resource string?

Reply:
We're saying "If the resource that we're requesting a token for is the same as our current resource or is a child (subpath) of it, then we're good."      I'll punch up the comment a bit to try and clarify.

Token Resource:  `amqps://somehub.servicebus.com/hub-name`
Good: `amqps://somehub.servicebus.com/hub-name/partitions/0`
Bad: `amqps://somehub.servicebus.com/different-hub`
Bad: `amqps://different.servicebus.com/hub-name`

Reply:
Rewrote and expanded the comment with examples.   I also overhauled the associated tests to be more aligned with the expected good/bad scenarios.